### PR TITLE
#15 angular guard

### PR DESCRIPTION
**Pull request by noklen**

### CODE REVIEW

File: src/app/interceptors/error.interceptor.ts [hunk 1]
```diff
@@ -16,9 +16,8 @@ export class ErrorInterceptor implements HttpInterceptor {
   public intercept(request: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
     return next.handle(request).pipe(catchError(err => {
       if (err.status === 401) {
-        // auto logout if 401 response returned from api
         this.authenticationService.signOut();
-        location.reload(true);
+        location.reload();
```
Comment (AlexanderBeichuk) [inline src/app/interceptors/error.interceptor.ts:20]: Why remove 'true'

File: src/app/services/auth.guard.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+  canActivate(
+    next: ActivatedRouteSnapshot,
+    state: RouterStateSnapshot): boolean {
+    const url: string = state.url;
+    return this.checkLogin(url);
+  }
+
+  checkLogin(url: string): boolean {
+    if (this.authService.isAuthorised() === true) { return true; }
```
Comment (AlexanderBeichuk) [inline src/app/services/auth.guard.ts:19]: isAuthorised rename to isAuthorized

File: src/app/services/authentication.service.ts [hunk 3]
```diff
@@ -27,6 +27,13 @@ export class AuthenticationService {
       });
   }
 
+  public isAuthorised(): boolean {
+    if (this.storageService.getStorageBearerToken()) {
+      return true;
+    }
+    return false;
+  }
+
```
Comment (AlexanderBeichuk) [inline src/app/services/authentication.service.ts:36]: remove this function

File: src/app/services/auth.guard.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    this.authService.redirectUrl = url;
+    this.router.navigate(['']);
+    return false;
+  }
+}
```
Comment (AlexanderBeichuk) [inline src/app/services/auth.guard.ts:24]: move content of this function to canActivate

File: src/app/services/authentication.service.ts [hunk 5]
```diff
@@ -5,17 +5,17 @@ import 'rxjs/add/operator/map';
 import {User} from '../models/user.model';
 import {Url} from '../models/url.enum';
 import {StorageService} from './storage.service';
-import {Router} from '@angular/router';
 
 @Injectable({
   providedIn: 'root'
 })
 export class AuthenticationService {
+  public redirectUrl: string;
```
Comment (AlexanderBeichuk) [inline src/app/services/authentication.service.ts:13]: What is it?

File: src/app/interceptors/error.interceptor.ts (hunk 1 above)
Comment (noklen) [reply src/app/interceptors/error.interceptor.ts:20]: diplicate

File: src/app/services/authentication.service.ts (hunk 5 above)
Comment (noklen) [reply src/app/services/authentication.service.ts:13]: done

File: src/app/route.ts [hunk 6]
```diff
@@ -8,6 +8,10 @@ import {AboutComponent} from './pages/about/about.component';
 import {TermsComponent} from './pages/terms/terms.component';
 import {PrivacyComponent} from './pages/privacy/privacy.component';
 import {ContactsComponent} from './pages/contacts/contacts.component';
+import {AuthGuard as AuthGuard} from './guards/auth.guard';
+import {RoleGuard as RoleGuard} from './guards/role.guard';
```
Comment (AlexanderBeichuk) [inline src/app/route.ts:12]: Why 'as RoleGuard' and 'as AuthGuard'

File: src/app/route.ts [hunk 7]
```diff
@@ -30,6 +34,22 @@ export const appRoutes: Routes = [
...
+    canActivate: [AuthGuard],
+    data: {
+      expectedRole: 'user'
+    }
+  },
+  {
+    path: 'admin',
+    component: AdminComponent,
+    canActivate: [AuthGuard],
+    resolve: {
```
Comment (AlexanderBeichuk) [inline src/app/route.ts:49]: use "data" instead "resolve"
RoleGuard instead AuthGuard

**Pull request merged**